pkg/cri/resource-manager/cache/affinity.go

```diff
@@ -178,11 +164,88 @@ func (e *Expression) Evaluate(container Container) bool {
 }
 
 // KeyValue extracts the value of the expresssion key from a container.
-func (e *Expression) KeyValue(container Container) (string, bool) {
-	value, ok := container.GetLabel(e.Key)
+func (e *Expression) KeyValue(c Container) (string, bool) {
+	value, ok, _ := c.(*container).resolveRef(e.Key)
 	return value, ok
 }
 
+// resolveRef walks an object trying to resolve a reference to a value.
+func (c *container) resolveRef(path string) (string, bool, error) {
+	var obj interface{}
+
+	c.cache.Debug("resolving %s/%s...", c.PrettyName(), path)
+
+	obj = c
+	ref := strings.Split(path, "/")
+	if len(ref) == 1 {
+		ref = []string{"labels", path}
+	}
+	for {
+		key := ref[0]
+		c.cache.Debug("* resolve: walking %s, @%s, obj %T...", path, key, obj)
+		switch obj.(type) {
```

Comment:
Could you use something like `switch v := obj.(type)` here? You should be able to avoid the type assertions below with that.

Reply:
If that works, then definitely. I'm just still such a golang n00b that I didn't know you can do that.

Reply:
Changed it to be like that.

Reply:
> If that works, then definitely. I'm just still such a golang n00b that I didn't know you can do that.

Ha, I'm the total n00b here. But it's comforting to see that even you don't know everything, yet 😄